### DISCUSSION TITLE
Read row & pass to REST API via POST

### DISCUSSION THREAD

**Discussion by anuragbhatia** [Budibase/budibase#9230] [Help] [answered]
I am trying to achieve following: 

Row update in Budibase DB > trigger automated action via "Automate" > Call REST API POST request and pass parameters of updated row. Somehow that is not working. 

My budibase db has multiple columns and one of that is "action". And that can have accept or reject. 
Now, in REST API Params I am using {{ triger.row.action }} but the updated row action is not being passed in the API call. 

Is that correct way of achieving this i.e form update > DB update > trigger REST API with updated data for an external source.

**Comment by melohagan** [comment]
You have a typo in trigger, should be `{{ trigger.row.action }}` .

**Reply by anuragbhatia** [reply]
Thanks for pointing that. I fixed it and still it doesn't work. The API call happens but the updated row entry is not passed in the REST API. 

Any suggestions on what am I missing on this?

**Reply by melohagan** [reply]
Hey Anurag, could you please provide some screenshots of your Automation steps

**Comment by anuragbhatia** [comment]
Sure, here's what I am trying to achieve in test case. 

1. New row added in a Budibase DB with three coloums - IP, type and action. The action is the one I am relying on in the test. 
![image](https://user-images.githubusercontent.com/1254255/211120556-8f1f5466-fef5-4c88-b4fe-3b66c075283b.png)

Here column name is "action" and can take value accept or reject. 
![image](https://user-images.githubusercontent.com/1254255/211120614-36f6b1b3-a134-4188-a25b-7d2b7f09c4ee.png)

<br/>

2. Automation runs on "row creation" and calls an external data connector which is Gitlab CI/CD pipeline REST API. Here's the screenshot of automation page: 
![image](https://user-images.githubusercontent.com/1254255/211120722-7381668b-a66b-424e-a897-89aaabbcc7ca.png)

<br/>

3. Here is the Gitlab REST API screenshot

![1](https://user-images.githubusercontent.com/1254255/211121368-f60b60e5-7373-4a1d-a8a9-29ea84871365.jpg)


Intention here is that value of ```action``` i.e ```accept``` or ```reject``` is passed to the REST API. This is where it is failing. If I put a static here, that works (confirming format of Gitlab CI/CD API). 



Thanks.

**Comment by melohagan** [comment, marked as answer]
Hey @anuragbhatia 

Thank you for the screenshots, that helped a lot. I believe you've misunderstood the bindings a little bit.

The `{{ trigger.row... }}` bindings can only be used within the Automations section. You need to expose a binding for the REST connector in the Bindings tab, and then pass through a value from the automation.

For example:
![Screenshot 2023-01-09 at 09 28 09](https://user-images.githubusercontent.com/101575380/211276861-7723f9e6-d011-4998-a5f7-be1ebddafa4c.png)

![Screenshot 2023-01-09 at 09 29 17](https://user-images.githubusercontent.com/101575380/211277070-aa21d7eb-95b6-4ee3-9859-ea3ad04cc886.png)

![Screenshot 2023-01-09 at 09 30 42](https://user-images.githubusercontent.com/101575380/211277322-e5f4badd-f0b3-42bb-9407-7737d054e9c2.png)

**Reply by anuragbhatia** [reply]
This works. Thanks a lot for your help.

**Reply by melohagan** [reply]
You're welcome :)

**Reply by anuragbhatia** [reply]
While I closed this issue, just one additional related question if you don't mind. What is right way to pass multiple variables using same mechanism? Multiple variables from same single row insert.  

So instead of just action, if I want var1, var2 etc, should I use {{ trigger.row }} in automation instead of hardcoding it to {{ trigger.row.action }}? 

How would binding look like in that case? 


Thanks again for your help.

**Reply by melohagan** [reply]
You could just add a binding for each variable in your REST connector, and then in your automation pass each one via `{{ trigger.row.var1 }}` etc
